pkg/cloudprovider/providers/azure/azure_controller_vmss.go

```diff
@@ -70,7 +80,7 @@ func (ss *scaleSet) AttachDisk(isManagedDisk bool, diskName, diskURI string, nod
 	defer ss.vmssVMCache.Delete(ss.makeVmssVMName(ssName, instanceID))
 
 	glog.V(2).Infof("azureDisk - update(%s): vm(%s) - attach disk(%s)", ss.resourceGroup, nodeName, diskName)
-	_, err = ss.VirtualMachineScaleSetVMsClient.Update(ctx, ss.resourceGroup, ssName, instanceID, vm)
+	_, err = ss.VirtualMachineScaleSetVMsClient.Update(ctx, ss.resourceGroup, ssName, instanceID, newVM)
```

Comment:
Adding PR just for future reference: https://github.com/kubernetes/kubernetes/pull/71992